pvlib/scaling.py

```diff
@@ -0,0 +1,320 @@
+        ignored otherwise.
+
+    density : numeric, default 41
+        The density of installed PV in W/m^2. Must be specified for 'square'
+        method, ignored otherwise. Default of 41 W/m^2 is 1MW per 6 acres.
+
+    dt : numeric
+        The time series time delta. By default, is inferred from the cs_series.
+        Must be specified for a time series that doesn't include an index.
+
+    Returns
+    -------
+    smoothed : pandas.Series or numeric, depending on inlet type
+        The clear sky index time series smoothed for the described plant.
+
+    wavelet: numeric
+        The individual wavelets for the time series before smoothing
+
+    tmscales: numeric
+        The timescales (in sec) associated with the wavelets
+
+    References
+    ----------
+    [1] M. Lave, J. Kleissl and J.S. Stein. A Wavelet-Based Variability
+    Model (WVM) for Solar PV Power Plants. IEEE Transactions on Sustainable
+    Energy, vol. 4, no. 2, pp. 501-509, 2013.
+
+    [2] M. Lave and J. Kleissl. Cloud speed impact on solar variability
+    scaling - Application to the wavelet variability model. Solar Energy,
+    vol. 91, pp. 11-21, 2013.
+
+    [3] Wavelet Variability Model - Matlab Code:
+    https://pvpmc.sandia.gov/applications/wavelet-variability-model/
+    """
+
+    # Added by Joe Ranalli (@jranalli), Penn State Hazleton, 2019
+
+    try:
+        import scipy.optimize
+    except ImportError:
+        raise ImportError("The WVM function requires scipy.")
+
+    dist = _compute_distances(longitude, latitude, method, capacity, density)
+    wavelet, tmscales = _compute_wavelet(cs_series, dt)
+
+    n_pairs = len(dist)
+    # Find eff length of position vector, 'dist' is full pairwise
+
+    def fn(x):
+        return np.abs((x ** 2 - x) / 2 - n_pairs)
+    n_dist = np.round(scipy.optimize.fmin(fn, np.sqrt(n_pairs), disp=False))
+
+    # Compute VR
+    A = cloud_speed / 2  # Resultant fit for A from [2]
+    vr = np.zeros(tmscales.shape)
+    for i, tmscale in enumerate(tmscales):
+        rho = np.exp(-1 / A * dist / tmscale)  # Eq 5 from [1]
+
+        # 2*rho is because rho_ij = rho_ji. +n_dist accounts for sum(rho_ii=1)
+        denominator = 2 * np.sum(rho) + n_dist
+        vr[i] = n_dist ** 2 / denominator  # Eq 6 of [1]
+
+    # Scale each wavelet by VR (Eq 7 in [1])
+    wavelet_smooth = np.zeros_like(wavelet)
+    for i in np.arange(len(tmscales)):
+        if i < len(tmscales) - 1:  # Treat the lowest freq differently
+            wavelet_smooth[i, :] = wavelet[i, :] / np.sqrt(vr[i])
+        else:
+            wavelet_smooth[i, :] = wavelet[i, :]
+
+    outsignal = np.sum(wavelet_smooth, 0)
+
+    try:  # See if there's an index already, if so, return as a pandas Series
+        smoothed = pd.Series(outsignal, index=cs_series.index)
+    except AttributeError:
+        smoothed = outsignal  # just output the numpy signal
+
+    return smoothed, wavelet, tmscales
+
+
+def _latlon_to_dist(latitude, longitude):
```

Comment:
I'd make this function public and name it `latlon_to_xy` perhaps

Reply:
The methodology here (from Matlab) is very much a "good enough" one to serve as internals of this function, but I don't think it would really be recommended for proper geographic distance calculation. Other packages exist to do it right. If the decision is to make wvm accept easting/northing pairs, and this is made public as helper so that those other packages remain optional, I think it would warrant notes as such in the docs.

Reply:
I agree. This adds support to switching from lat, lon input to x,y.  The `_latlon_to_dist` function is an added convenience, but not a necessity. Because the WVM is likely being used over relatively small distances (10s of km) I'm not overly concerned about the numerical accuracy here - should be good enough.

Reply:
My vote is change inputs to discrete pairs of `(x_east, y_north)` and either discard this function, or as @cwhanse suggests make it public, move it to `tools`, add disclaimer/warning Cliff suggests: _only over short distances_, and provide links to [`pyproj.transform()`](https://pyproj4.github.io/pyproj/dev/api/transformer.html#pyproj-transform) and other more accurate projection methods

Reply:
I don't really consider `tools` to be part of the public API even though it's not explicitly private. I can't think of any instance in which we recommend that users directly call a `tools` function. I recommend keeping it here unless we know of another instance in which pvlib itself would make use of this function. 

In any case, I agree with earlier comments to require distances instead of coordinates, and I don't actually think pvlib needs this function at all.

Reply:
> In any case, I agree with earlier comments to require distances instead of coordinates, and I don't actually think pvlib needs this function at all.

@wholmgren Are you OK with x,y coordinate pairs as input? Although the coordinates are only used to compute distances, asking for distances as input is less intuitive to me.

Reply:
Sorry.. yes, updated my unclear comment above.